src/emitters/distant.cpp

```diff
@@ -0,0 +1,157 @@
+
+            m_world_transform =
+                new AnimatedTransform(ScalarTransform4f::look_at(
+                    ScalarPoint3f(0.0f), ScalarPoint3f(direction), up));
+        }
+
+        m_flags      = EmitterFlags::Infinite | EmitterFlags::DeltaDirection;
+        m_irradiance = props.texture<Texture>("irradiance", Texture::D65(1.f));
+    }
+
+    void set_scene(const Scene *scene) override {
+        m_bsphere = scene->bbox().bounding_sphere();
+        m_bsphere.radius =
+            max(math::RayEpsilon<Float>,
+                m_bsphere.radius * (1.f + math::RayEpsilon<Float>) );
+    }
+
+    Spectrum eval(const SurfaceInteraction3f &si, Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointEvaluate, active);
+        Vector3f v = m_world_transform->eval(si.time, active)
+                         .inverse()
+                         .transform_affine(-si.wi);
+        return (allclose(v, Vector3f{ 0.f, 0.f, 1.f }))
+                   ? unpolarized<Spectrum>(m_irradiance->eval(si, active))
+                   : unpolarized<Spectrum>(0.f);
+    }
+
+    std::pair<Ray3f, Spectrum> sample_ray(Float /*time*/,
+                                          Float /*wavelength_sample*/,
+                                          const Point2f & /*sample2*/,
+                                          const Point2f & /*sample3*/,
+                                          Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointSampleRay, active);
+        NotImplementedError("sample_ray");
+    }
+
+    std::pair<DirectionSample3f, Spectrum>
+    sample_direction(const Interaction3f &it, const Point2f & /*sample*/,
+                     Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointSampleDirection, active);
+
+        Vector3f d = m_world_transform->eval(it.time, active)
+                         .transform_affine(Vector3f{ 0.f, 0.f, 1.f });
+        Float dist = 2.f * m_bsphere.radius;
+
+        DirectionSample3f ds;
+        ds.p      = it.p - d * dist;
+        ds.n      = d;
+        ds.uv     = Point2f(0.f);
+        ds.time   = it.time;
+        ds.pdf    = 1.f;
+        ds.delta  = true;
+        ds.object = this;
+        ds.d      = -d;
+        ds.dist   = dist;
+
+        SurfaceInteraction3f si = zero<SurfaceInteraction3f>();
+        si.wavelengths          = it.wavelengths;
+
+        return std::make_pair(
+            ds, unpolarized<Spectrum>(m_irradiance->eval(si, active)) / ds.pdf);
+    }
+
+    Float pdf_direction(const Interaction3f &, const DirectionSample3f &ds,
+                        Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::EndpointEvaluate, active);
+
+        Vector3f d = m_world_transform->eval(ds.time, active)
+                         .transform_affine(Vector3f{ 0.f, 0.f, 1.f });
+        return allclose(ds.d, -d) ? 1.0f : 0.0f;
+    }
```

Comment:
Should always return `0.f` as it is a `DeltaDirection` light.

Reply:
I also got this from the Mitsuba 1 implementation; should I change it?

Reply:
It's fixed.